source/common/stats/custom_stat_namespaces_impl.cc

```diff
@@ -0,0 +1,25 @@
+#include "source/common/stats/custom_stat_namespaces_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+bool CustomStatNamespacesImpl::registered(const absl::string_view& name) const {
+  return namespaces_.find(name) != namespaces_.end();
+}
+
+void CustomStatNamespacesImpl::registerStatNamespace(const absl::string_view& name) {
+  namespaces_.insert(std::string(name));
+};
+
+absl::optional<std::string>
+CustomStatNamespacesImpl::trySanitizeStatName(const absl::string_view& stat_name) const {
```

Comment:
Looking at what this API actually does, I think it might be more accurate to rename it `stripRegisteredPrefix`. It's doing no other sanitization.